core/src/test/scala/org/http4s/blaze/channel/nio2/ClientChannelFactorySpec.scala

```diff
@@ -0,0 +1,32 @@
+package org.http4s.blaze.channel.nio2
+
+import java.net.{InetSocketAddress, SocketTimeoutException}
+
+import org.http4s.blaze.util.TickWheelExecutor
+import org.specs2.mutable.{After, Specification}
+
+import scala.concurrent.Await
+import scala.concurrent.duration.DurationDouble
+
+class ClientChannelFactorySpec extends Specification {
+
+  "ClientChannelFactory" should {
+    "time out" in new fastTickWheelExecutor {
+      val factory =
+        new ClientChannelFactory(connectingTimeout = Some(1.millisecond), scheduler = scheduler)
+      val address = new InetSocketAddress("1.1.1.1", 1)
```

Comment:
That address is a CloudFlare service.  It doesn't listen on port 1, so it works, and probably always will, but I'm always reluctant to send traffic to servers that aren't ours. IPv6 [reserves `0100::/64` as a blackhole](https://tools.ietf.org/html/rfc6666).  I don't know of anything similar in IPv4, though there are a few [test nets](https://tools.ietf.org/html/rfc5737) that _probably_ behave similarly.